### ISSUE TITLE
Migrate away from AngularJS

### ISSUE THREAD

**Issue by e-alfred** [nextcloud/weather#80]
AngularJS is in LTS mode currently and will be deprecated in 2021 after a 3 year period. We should move away from AngularJS to Vue probably at some point in the future.

**Comment by nerzhul** [comment]
@e-alfred i approve totally with you. the move has been done on ocsms if i remember but not on this. It's not difficult to move away from angular as the templating may be the same. Also the new code may be proper than before :)

**Comment by berdosi** [comment]
Started to fool around with this issue, but there's still a lot of work to do there:
https://github.com/berdosi/weather/tree/migrate-to-vue

There are still bugs, missing features, and lots of old code abused and patched up for reuse. It more of a demo, than a meaningful contribution at the moment.

**Comment by e-alfred** [comment]
@berdosi Nice work for a start. It isn't quite urgent as there is still support for AngularJS until 2021, but it would be good to do it until then anyway. Maybe you can make a draft PR and work from there onwards? We could merge it the if it becomes stable sometime in the future.

**Comment by nerzhul** [comment]
hello @berdosi i looked at your MR seems promising but not very readable JS code unfortunately due to the one big file :(

**Comment by berdosi** [comment]
Hello, @nerzhul, I'll make efforts to structure it better for the to-be-merged version; it should be relatively easy to split the code to components (ocsms is a good point of reference :)).

I still have some warnings to clean up before starting to refactor, though.

**Comment by berdosi** [comment]
I split it up to Settings / CityList / Forecast, and the components communicate with each other through a shared store. I don't consider it to be very idiomatic, but at least the file is split up.
(While app.js is still in place for my reference, it isn't linked to through main.php)

Apart from [migrate-to-vue](https://github.com/berdosi/weather/tree/migrate-to-vue), I've started playing with  [migrate-to-vue-split-components](https://github.com/berdosi/weather/tree/migrate-to-vue-split-components/js), using separate templates. I consider the other version with fewer, bigger files more readable.
